utils/eth1.go

```diff
@@ -346,6 +347,110 @@ func formatAmount(amount *big.Int, unit string, digits int, maxPreCommaDigitsBef
 	return template.HTML(fmt.Sprintf("<span%s>%s%s</span>", tooltip, preComma, displayUnit))
 }
 
+// NewFormat returns a new HTML-formatted string representing the given Wei big.Int amount, with the specified number of digits after the decimal point and number of digits hidden on mobile views.
+//
+// Parameters:
+//   - `amount` : the Wei(!) amount to be formatted.
+//   - `unit` : the unit in which the amount should be formatted. Supported values are ETH and GWei.
+//   - `digits` : the number of digits after the decimal point to be displayed.
+//   - `mobileDigitsHide` : the number of trailing decimal digits to hide on mobile views.
+//   - `maxPreCommaDigitsBeforeTrim` : the maximum number of digits before the decimal point that should be displayed before trimming excess decimal digits. If this is set to 0, no digits are trimmed.
+//
+// **NOTE** Every number is rounded down
+//
+// TODO Replace every formatAmount with this
+func NewFormat(amount *big.Int, unit string, digits int, mobileDigitsHide int, maxPreCommaDigitsBeforeTrim int) template.HTML {
```

Comment:
This should most likely be named something like **FormatWeiAmount**.

Reply:
My goal was to replace the formatAmount function, once every call of formatAmount has been replaced with NewFormat over the course of implementing all new designs I will change the function name